### DISCUSSION TITLE
How to render primitive / shape index maps without antialiasing?

### DISCUSSION THREAD

**Discussion by ShnitzelKiller** [mitsuba-renderer/mitsuba3#661] [Q&A] [answered]
The docs [mention ](https://mitsuba.readthedocs.io/en/stable/src/generated/plugins_integrators.html#arbitrary-output-variables-integrator-aov) that the value of the prim_index and shape_index are meaningless with fractional pixel coverage. Is there some way to ensure that this integrator does not antialias / supersample at all, so that the output is purely integer values that can be trusted? I tried setting spp=1, and have no reconstruction filters, but there are still fractional valued pixels.

![image](https://user-images.githubusercontent.com/6132502/231903143-eec2d6d3-4594-49d9-b2f9-a654a2aaf78f.png)

**Comment by njroussel** [comment, marked as answer]
Hi @ShnitzelKiller 

Specifying no reconstruction filter will default to a `gaussian` filter (see [here](https://mitsuba.readthedocs.io/en/stable/src/generated/plugins_films.html#high-dynamic-range-film-hdrfilm)). You want to use a [`box`](https://mitsuba.readthedocs.io/en/stable/src/generated/plugins_rfilters.html#box-filter-box) filter to get such a result.
